rasa/core/agent.py

```diff
@@ -408,7 +408,7 @@ def load(
             if not os.path.exists(model_path):
                 raise ModelNotFound(f"No file or directory at '{model_path}'.")
             if os.path.isfile(model_path):
-                model_path = get_model(model_path)
+                model_path = get_model(str(model_path))
```

Comment:
The type annotation for the `model_path` argument needs to be updated

Reply:
We have checked the signatures of the related functions (```load()``` and ```get_model()```) their type annotations seem to be correct, so we didn't understand the suggested modification. Could you please clarify?

Reply:
sure - casting the model path to `str` would only be necessary if `model_path` is of type `pathlib.Path`. the type annotations in the `load` function only have `model_path: Text`

Reply:
We added type annotation to `pathlib.Path` and added type checking. When you can, check if the changes are correct, please.